integration/firestore/bundle_test_data/generate_test_bundle.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * The main function for this script.
+ * @param {array} args The command line arguments.
+ */
+import * as firestore from '@firebase/firestore-types';
+import { TestBundleBuilder } from '@firebase/firestore/test/unit/util/bundle_data';
+import { DatabaseId } from '@firebase/firestore/src/core/database_info';
+import { key } from '@firebase/firestore/test/util/helpers';
+import { collectionReference } from '@firebase/firestore/test/util/api_helpers';
+import {
+  EMULATOR_PROJECT_ID,
+  ALT_PROJECT_ID
+} from '@firebase/firestore/test/integration/util/emulator_settings';
+const PROJECT_CONFIG = require('../../../../config/project.json');
+
+/**
+ * Returns a bundle builder for the given projectId. The builder will build with the
+ * same documents and queries, but with the given projectId.
+ */
+function bundleWithTestDocsAndQueries(projectId: string): TestBundleBuilder {
+  const testDocs: { [key: string]: firestore.DocumentData } = {
+    a: { k: { stringValue: 'a' }, bar: { integerValue: 1 } },
+    b: { k: { stringValue: 'b' }, bar: { integerValue: 2 } }
+  };
+
+  const a = key('coll-1/a');
+  const b = key('coll-1/b');
+  const builder = new TestBundleBuilder(new DatabaseId(projectId));
+
+  builder.addNamedQuery(
+    'limit',
+    { seconds: 1000, nanos: 9999 },
+    (collectionReference('coll-1')
+      .orderBy('bar', 'desc')
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      .limit(1) as any)._query
```

Comment:
Should `TestBundleBuilder` take a public API query? It should then be able to cast internally to the internal query.

Reply:
It is shared with spec tests, and I think it would help those to keep using the internal one?